src/mscorlib/src/System/BitConverter.cs

```diff
@@ -339,21 +339,25 @@ public static bool ToBoolean(byte[] value, int startIndex)
             return value[startIndex] != 0;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] // This should compile down to nothing or roughly 1 instruction, depending on the register holding value
```

Comment:
If forced to not inline on `BitConverter.DoubleToInt64Bits` and on `double.IsNaN`, the codegen is:

```asm
sub         rsp,28h
vzeroupper
vmovsd      xmm0,qword ptr [7FFD436104F0h]     ; double.IsNaN(double.NaN)
call        00007FFD43610090                   ;
```

**double.IsNaN**
```asm
sub         rsp,28h
vzeroupper
call        00007FFD43610420                   ; bits = BitConverter.DoubleToInt64Bits(d)
mov         rdx,7FFFFFFFFFFFFFFFh              ; bits &= 0x7FFFFFFFFFFFFFFF
and         rax,rdx                            ;
mov         rdx,7FF0000000000000h              ; return bits > 0x7FF0000000000000
cmp         rax,rdx                            ;
setg        al                                 ;
movzx       eax,al                             ;
add         rsp,28h
ret
```

**BitConverter.DoubleToInt64Bits**
```asm
vmovsd      qword ptr [rsp+8],xmm0             ; return *((long*)&value)
mov         rax,qword ptr [rsp+8]              ;
ret
```

If allowing inlining on `BitConverter.DoubleToInt64Bits` but forced to not inline on `double.IsNaN`, the codegen is:

```asm
sub         rsp,28h
vzeroupper
vmovsd      xmm0,qword ptr [7FFD436104A8h]     ; Load double.NaN
call        00007FFD43610090                   ; Call double.IsNaN
```

**double.IsNaN**
```asm
push        rax                                ;
vzeroupper                                     ;
vmovsd      qword ptr [rsp],xmm0               ; bits = *((long*)&value)
mov         rax,7FFFFFFFFFFFFFFFh              ; bits &= 0x7FFFFFFFFFFFFFFF
and         rax,qword ptr [rsp]                ;
mov         rdx,7FF0000000000000h              ; return bits > 0x7FF0000000000000
cmp         rax,rdx                            ;
setg        al                                 ;
movzx       eax,al                             ;
add         rsp,8
ret  
```

If allowing inlining on both, the codegen is:
```asm
sub         rsp,28h
vzeroupper
vmovsd      xmm0,qword ptr [7FFD435F04D0h]     ; bits *((long*)&value)
vmovsd      qword ptr [rsp+20h],xmm0           ;
mov         rcx,7FFFFFFFFFFFFFFFh              ; bits &= 0x7FFFFFFFFFFFFFFF
and         rcx,qword ptr [rsp+20h]            ;
mov         rax,7FF0000000000000h              ; result = bits > 0x7FF0000000000000
cmp         rcx,rax                            ;
setg        cl                                 ;
movzx       ecx,cl                             ;
```